### PR TITLE
[s] Invisimin HUD Fix + Changeling AI HUD Fix + Fixes Changeling Digital Camo with AI

### LINKED ISSUE #22040: AI's sechud still shows on targets that have digital camo
Was playing AI a few weeks ago when I saw a changeling using digital camo to hide from cams, when I switched to sechud mode I was able to see him running around and even track him to an extent from his job icon/wanted status.

### LINKED ISSUE #36709: Invisimin doesn't stop HUDs from seeing you
Invisimin is a valuable tool for moving mobs around in events, but SecHUDs can still see invisible admins which defeats the purpose entirely.

### LINKED ISSUE #36759: Changeling Digital Camo Exploit
Tested with the current repo.

When changeling uses digital camo, it goes invisible to the AI. But when the changeling toggles off the digital camo, he is still invisible to the AI. Also #22040 is still broken.

1. Become an AI as your 1st client
2. Become a changeling as your 2nd client
3. Get digital camo upgrade from changeling
4. Turn it on, you'll see the changeling go invisible with your AI client.
5. Turn it off, the AI client will still see the changeling client as invisible.

### PR DESCRIPTION
**Pull request by Davidj361**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl:
fix: Fixed admins showing up on huds when invisimin
fix: AI is no longer able to see changeling with its hud when changeling has digital camo on.
fix: AI can now see changeling when changeling disables its digital camo.
/:cl:

Fixes #36709
Fixes #22040
Fixes #36759

NOTE: I think there needs to be more testing with various cases like taking on & off the sec glasses, spawning in players, antags, and whatever.

### CODE REVIEW

**Comment by Davidj361** [comment]
I'll do the changeling hud thing in a different PR, there a lot of broken crap.

File: code/datums/hud.dm [hunk 1]
```diff
@@ -106,6 +106,10 @@ GLOBAL_LIST_INIT(huds, list(
 /datum/atom_hud/proc/add_to_single_hud(mob/M, atom/A) //unsafe, no sanity apart from client
 	if(!M || !M.client || !A)
 		return
+	if (ismob(A))
+		var/mob/m = A
+		if (m.digitalinvis && isAI(M))
```
Comment (ShizCalev) [inline code/datums/hud.dm]: I'm not too crazy about having both M & m in the same line. Could you change it to something else? lol

**Review by ShizCalev — DISMISSED** [review]

File: code/datums/hud.dm (hunk 1 above)
Comment (Davidj361) [reply code/datums/hud.dm]: > using uppercase local variable names

**Comment by Davidj361** [comment]
Done

**Comment by Davidj361** [comment]
Also, I feel that `reload_huds_of_me()` has a lot of overhead for just trying to reload the huds of AI. However, I wanted to play it safe by reloading EVERYONE'S huds. Let me know if you think I should change that.

And
```
mob.add_to_all_human_data_huds()
mob.mind.invisi_join_antag_huds()
```
possibly has overhead, but I believe it is needed in order to disappear from all huds when invising.

**Comment by optimumtact** [comment]
did you get more testing?

**Comment by ninjanomnom** [comment]
Ping for reopen if you get back to working on this

**Comment by Davidj361** [comment]
I'm noticing a problem with the merge as fixing the invisibility of changeling on camo disable utilized huds apparently, so I need to make some adjustments.

**Comment by Davidj361** [comment]
Ok, I tested the invisimin and the changeling stuff. Should be good to merge.

**Comment by Davidj361** [comment]
The alternative appearance and disabling of data huds for the ling doesn't work when new AI are spawned in.

**Comment by optimumtact** [comment]
alternate apperances are still broken @AnturK

**Comment by AlexTheSysop42** [comment]
Is that why the cardborg suit and ai hologram don't change direction or should I actually get around to reporting that?

**Comment by AnturK** [comment]
Broken how, from my testing the only thing that's broken is this PR.

**Comment by optimumtact** [comment]
they still dont' appear to late joiners

**Comment by optimumtact** [comment]
either that or my testing is wrong

**Comment by AlexTheSysop42** [comment]
Cardborg suit, even when visible, doesn't turn and thus is completely useless for any actual disguising

**Comment by optimumtact** [comment]
it's not for disguising, it's for pretending to be a borg with your friends

**Comment by AnturK** [comment]
I'm going to close this PR for now, ping when you want to continue working on it. And if anyone has any reproducible AA bugs please put up an issue report.

**Pull request closed without merging**